ludwig/utils/tokenizers.py

```diff
@@ -1151,27 +1174,59 @@ def get_hf_tokenizer(pretrained_model_name_or_path, **kwargs):
     Returns:
         A torchscript-able HF tokenizer if it is available. Else, returns vanilla HF tokenizer.
     """
+    from transformers import AutoTokenizer, BertTokenizer
+
+    hf_name = pretrained_model_name_or_path
+    # use_fast=False to leverage python class inheritance
+    # cannot tokenize HF tokenizers directly because HF lacks strict typing and List[str] cannot be traced
+    hf_tokenizer = AutoTokenizer.from_pretrained(hf_name, use_fast=False)
+
+    torchtext_tokenizer = None
+    if "bert" in TORCHSCRIPT_COMPATIBLE_TOKENIZERS and isinstance(hf_tokenizer, BertTokenizer):
+        tokenizer_kwargs = _get_bert_config(hf_name)
+        torchtext_tokenizer = BERTTokenizer(
+            **tokenizer_kwargs,
+            is_hf_tokenizer=True,
+            hf_tokenizer_attrs={
+                "pad_token": hf_tokenizer.pad_token,
+                "unk_token": hf_tokenizer.unk_token,
+                "cls_token_id": hf_tokenizer.cls_token_id,
+                "sep_token_id": hf_tokenizer.sep_token_id,
+                "all_special_tokens": hf_tokenizer.all_special_tokens,
+            },
+        )
 
-    if "bert" in TORCHSCRIPT_COMPATIBLE_TOKENIZERS:
-        from transformers.models.bert.tokenization_bert import PRETRAINED_INIT_CONFIGURATION, PRETRAINED_VOCAB_FILES_MAP
+    use_torchtext = torchtext_tokenizer is not None
+    if use_torchtext:
+        # If a torchtext tokenizer is instantiable, tenatively we will use it. However,
+        # if the tokenizer does not pass (lightweight) validation, then we will fall back to the vanilla HF tokenizer.
+        # TODO(geoffrey): can we better validate tokenizer parity before swapping in the TorchText tokenizer?
+        # Samples from https://github.com/huggingface/transformers/blob/main/tests/models/bert/test_tokenization_bert.py
+        for sample_input in ["UNwant\u00E9d,running", "ah\u535A\u63A8zz", " \tHeLLo!how  \n Are yoU? [UNK]"]:
+            hf_output = hf_tokenizer.encode(sample_input)
+            tt_output = torchtext_tokenizer(sample_input)
+            if hf_output != tt_output:
+                use_torchtext = False
+                logging.warning("Falling back to HuggingFace tokenizer because TorchText tokenizer failed validation.")
```

Comment:
Ah, clarification: For a while I thought that the only reason why we didn't do global logging was because of a hack that was needed for tensorflow.

However, since then, I've empirically observed that named module logging produces better logs in ray runs, notebooks (or perhaps both? Would need to double check).

@arnavgrg may have additional thoughts here. At the moment, I believe we are in favor of going back to named module logging everywhere. Hope that makes sense

Reply:
Got it, thank you!